src/xml_document.cc

```diff
@@ -266,13 +266,28 @@ XmlDocument::New(xmlDoc* doc)
     return scope.Escape(obj);
 }
 
+char* getStringOption(v8::Local<v8::Value> value, const char* defaultValue = NULL) {
+    if (value->IsString()) {
+        v8::String::Utf8Value string(value);
+        char *str = (char *) malloc(string.length() + 1);
```

Comment:
This was code from my branch that consolidates the fromHTML and fromXML functions. I put it in here to try to track down some Window's node-gyp errors, but I think it's better off as a separate PR. I'm going to revert xml_document to the original as long as everything still builds.

I'll get the consistency issues fixed and we can review the other aspects in a separate PR. I think I saw in the source that libxml xmlFree()'s the `encoding` and `base_url` strings after it parses, but I'll have to look again.